fem/fe_coll.cpp

```diff
@@ -1927,82 +1981,87 @@ L2_FECollection::L2_FECollection(const int p, const int dim, const int btype,
          snprintf(d_name, 32, "%s_T%d_%dD_P%d", prefix, btype, dim, p);
    }
 
+   InitOrder(default_p);
+}
+
+
+bool L2_FECollection::HaveOrder(int p) const
+{
+   const auto &array = L2_Elements[Geometry::POINT];
```

Comment:
This function has some problems. First, `L2_Elements[Geometry::POINT]` is not even initialized unless `InitOrder` has been called for `dim = 0`. The constructor should initialize all entries of `L2_Elements` to `NULL`. Second, `L2_Elements[Geometry::POINT]` is set by `InitOrder` only for `dim=0`. Consequently, `InitOrder` gets called multiple times for `dim > 0`, resulting in memory leaks. The array to be checked should depend on the dimension. The same issues are also in `H1_FECollection::HaveOrder`.

Reply:
Hi Dylan, the treatment of variable order collections in this branch is obsolete. I will eventually be merging in the branch `var-order-space-dev`, where the handling of variable order is much simplified. You may actually want to switch to using `var-order-space-dev` which is more mature and will hopefully be merged into `master` soon.